File: clase5/codigo_jgrasp/all/DoublyNode.java
public class DoublyNode extends Node {
    
    // Atributo adicional para apuntar al nodo previo
    Node prev;

    public DoublyNode() {
        super(); 
        this.prev = null;
    }

    public DoublyNode(int data) {
        super(data); 
        this.prev = null;
    }
}

File: clase6/linded_list/doubly_LL/con_herencia/python/src/doubly_node.py
from node import Node

class DoublyNode(Node):
    def __init__(self, data=0):
        super().__init__(data)
        self.prev = None

    def __str__(self):
        return f"[{self.data}]"

if __name__ == "__main__":
    # Example usage
    node1 = DoublyNode(5)
    print(node1)  # Output: [5]
    node2 = DoublyNode()
    print(node2)  # Output: [0]
